Public/Src/Engine/Scheduler/Tracing/Log.cs

```diff
@@ -1210,6 +1210,15 @@ internal static string PipWriteFileFailedMessage(string pipDescription, string p
             Message = Events.PipPrefix + "The number of messages sent by detoured processes did not match the number received by the {MainExecutableName} process. Refer to the {ShortProductName} log for more information.")]
         public abstract void LogMismatchedDetoursErrorCount(LoggingContext context, long pipSemiStableHash, string pipDescription);
 
+        [GeneratedEvent(
+            (int)EventId.PipExitedWithDeadExitCode,
+            EventGenerators = EventGenerators.LocalOnly,
+            EventLevel = Level.Error,
+            Keywords = (int)Events.Keywords.UserMessage,
+            EventTask = (int)Events.Tasks.PipExecutor,
+            Message = Events.PipPrefix + "Pip exited with 0xDEAD exit code. Refer to the {ShortProductName} log for more information.")]

```

Comment:
So is the pip going to be retried? Could you make the message more explicit about what's going to happen? If you do retry, shouldn't this be a warning instead of an error?

Reply:
This log is only used after we perform several retries. Currently, with that exit code, we produces an error. So, this will also gives an error.